sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -244,7 +244,11 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
                 paths = rootPath.toString :: Nil,
                 userSpecifiedSchema = Option(updatedTable.dataSchema),
                 bucketSpec = None,
-                options = options,
+                // Do not interpret the 'path' option at all when tables are read using the Hive
+                // source, since the URIs will already have been read from the table's LOCATION.
+                // The `Map() ++` is necessary to force materialization since the return of
+                // `filterKeys` is a view which is not serializable
+                options = Map() ++ options.filterKeys(!_.equalsIgnoreCase("path")),
```

Comment:
I agree, `path` is a special data source option in Spark. For hive tables not written by Spark, it's wrong to apply the Spark knowledge here and treat the `path` option specially.

Reply:
nit: `options.filterKeys(!_.equalsIgnoreCase("path")).toMap`?

Reply:
Unfortunately `toMap` is insufficient to materialize the view, check out the Scaladoc/implementation:
```
  /** Returns this $coll as an immutable map.
   *
   *  A new map will not be built; lazy collections will stay lazy.
   */
  @deprecatedOverriding("Immutable maps should do nothing on toMap except return themselves cast as a map.",  "2.11.0")
  override def toMap[T, U](implicit ev: (K, V) <:< (T, U)): immutable.Map[T, U] =
    self.asInstanceOf[immutable.Map[T, U]]
```

Reply:
how about `options.filter { case (k, _) => !k.equalsIgnoreCase("path") }`?

Reply:
Done